api/app/Builders/PoolCandidateBuilder.php

```diff
@@ -0,0 +1,565 @@
+     */
+    public function whereHasPoolCandidateCommunity(?string $communityId): self
+    {
+        if (empty($communityId)) {
+            return $this;
+        }
+
+        return $this->whereHas('pool', function ($query) use ($communityId) {
+            $query->where('community_id', $communityId);
+        });
+    }
+
+    /**
+     * Scopes the query to return PoolCandidates in a pool with one of the specified classifications.
+     * If $classifications is empty, this scope will be ignored.
+     *
+     * @param  array|null  $classifications  Each classification is an object with a group and a level field.
+     */
+    public function whereAppliedClassificationsIn(?array $classifications): self
+    {
+        if (empty($classifications)) {
+            return $this;
+        }
+
+        return $this->whereHas('pool', function ($query) use ($classifications) {
+            $query->whereClassifications($classifications);
+        });
+
+    }
+
+    /**
+     * Scopes the query to only return PoolCandidates who are available in a pool with one of the specified classifications.
+     * If $classifications is empty, this scope will be ignored.
+     *
+     * @param  array|null  $classifications  Each classification is an object with a group and a level field.
+     */
+    public function whereQualifiedClassificationsIn(?array $classifications): self
+    {
+        if (empty($classifications)) {
+            return $this;
+        }
+
+        return $this->whereAvailable()
+            ->whereAppliedClassificationsIn($classifications);
+    }
+
+    /**
+     * Scope Publishing Groups
+     *
+     * Restrict a query by specific publishing groups
+     */
+    public function wherePublishingGroupsIn(?array $publishingGroups): self
+    {
+        if (empty($publishingGroups)) {
+            return $this;
+        }
+
+        return $this->whereHas('pool', function (Builder $query) use ($publishingGroups) {
+            /** @var \App\Builders\PoolBuilder $query */
+            $query->publishingGroups($publishingGroups);
+        });
+    }
+
+    /**
+     * Scope is IT & OTHER Publishing Groups
+     *
+     * Restrict a query by pool candidates that are for pools
+     * containing IT and OTHER publishing groups
+     */
+    public function whereInTalentSearchablePublishingGroup(): self
+    {
+        return $this->wherePublishingGroupsIn([
+            PublishingGroup::IT_JOBS->name,
+            PublishingGroup::OTHER->name,
+        ]);
+
+    }
+
+    public function whereOperationalRequirementsIn(?array $operationalRequirements): self
+    {
+        if (empty($operationalRequirements)) {
+            return $this;
+        }
+
+        return $this->whereHas('user', function ($query) use ($operationalRequirements) {
+            $query->whereOperationalRequirementsIn($operationalRequirements);
+        });
```

Comment:
I like the comment/pattern above. You can infer from the `whereHas` what model is being acted upon, but the comment makes it obvious and more importantly, where to find the method

![image](https://github.com/user-attachments/assets/be9b35ed-07de-4d34-9ab2-b50238d30f44)

Reply:
That comment is only necessary because PHPStan cannot infer unless it is prefixed with `where` and I'd love to remove it eventually :sweat_smile: